File: Bank/CSharp/src/Controllers/HomeController.fs
namespace Criipto.NugetSample.Controllers

open System.Diagnostics
open Microsoft.AspNetCore.Authorization

open Microsoft.AspNetCore.Mvc
open Microsoft.Extensions.Logging

open Criipto.NugetSample.Models

type HomeController (logger : ILogger<HomeController>) =
    inherit Controller()

    
    member this.Index () =
        this.View()
        
    [<Authorize>]
    member this.Overview(string accountName) =
        this.View()

    [<Authorize>]
    member this.Transactions(string accountName) =
        this.View()

    [<Authorize>]
    member this.Messages(string accountName) =
        this.View()

    [<ResponseCache(Duration = 0, Location = ResponseCacheLocation.None, NoStore = true)>]
    member this.Error () =
        let reqId = 
            if isNull Activity.Current then
                this.HttpContext.TraceIdentifier
            else
                Activity.Current.Id

        this.View({ RequestId = reqId })

File: Verify/dotnet-core-v6.x/Controllers/HomeController.cs
using System.Diagnostics;
using Microsoft.AspNetCore.Mvc;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Authentication;
using Microsoft.AspNetCore.Authentication.Cookies;
using Microsoft.AspNetCore.Authentication.OpenIdConnect;
using dotnet_core_v6.x.Models;

namespace dotnet_core_v6.x.Controllers;

public class HomeController : Controller
{
    private readonly ILogger<HomeController> _logger;

    public HomeController(ILogger<HomeController> logger)
    {
        _logger = logger;
    }

    public IActionResult Index()
    {
        return View();
    }
    
    [Authorize] // If not already authenticated this kicks off the process
    public IActionResult Protected()
    {
        return View();
    }

    public IActionResult Privacy()
    {
        return View();
    }

    public async Task Logout()
    {
        await HttpContext.SignOutAsync(OpenIdConnectDefaults.AuthenticationScheme);
        await HttpContext.SignOutAsync(CookieAuthenticationDefaults.AuthenticationScheme);
    }

    [ResponseCache(Duration = 0, Location = ResponseCacheLocation.None, NoStore = true)]
    public IActionResult Error()
    {
        return View(new ErrorViewModel { RequestId = Activity.Current?.Id ?? HttpContext.TraceIdentifier });
    }
}
